src/Libraries/Microsoft.Extensions.AI/ChatCompletion/ChatClientStructuredOutputExtensions.cs

```diff
@@ -44,16 +35,14 @@ public static partial class ChatClientStructuredOutputExtensions
     /// by the client, including any messages for roundtrips to the model as part of the implementation of this request, will be included.
     /// </remarks>
     /// <typeparam name="T">The type of structured output to request.</typeparam>
-    [RequiresUnreferencedCode(UsesReflectionJsonSerializerMessage)]
-    [RequiresDynamicCode(UsesReflectionJsonSerializerMessage)]
     public static Task<ChatCompletion<T>> CompleteAsync<T>(
         this IChatClient chatClient,
         IList<ChatMessage> chatMessages,
         ChatOptions? options = null,
         bool? useNativeJsonSchema = null,
         CancellationToken cancellationToken = default)
         where T : class =>
-        CompleteAsync<T>(chatClient, chatMessages, DefaultJsonSerializerOptions, options, useNativeJsonSchema, cancellationToken);
+        CompleteAsync<T>(chatClient, chatMessages, JsonDefaults.Options, options, useNativeJsonSchema, cancellationToken);
```

Comment:
I'm curious how this works without needing any annotations for AOT.

Does `JsonDefaults.Options.GetJsonSchemaAsNode` work with all possible user types in an AOT application?

Reply:
No, just the ones defined in the context. The idea being that if a type is missing it will unambiguously throw `NotSupportedException` as opposed to entering the undefined behaviour of the reflection-based serializer.

Reply:
> Or is it that in non-AOT app, `JsonDefaults.Options` enables the reflection-based serializer so it will work the same as before this change?

Exactly that. The constructor for `JsonDefault.Options` effectively checks if `PublishTrimmed` has been enabled for the application and either returns a reflection-based or source generated contract resolver:

https://github.com/dotnet/extensions/blob/9ea3e7614ca9bc48215aead3bd2d5cf6169309a0/src/Libraries/Microsoft.Extensions.AI/JsonDefaults.cs#L28-L44

This strategy reflects how aspnetcore configures its own `JsonSerializerOptions`

Reply:
Awesome - sounds great. Thanks!